src/nvim/os/pty_proc_unix.c

```diff
@@ -47,12 +47,17 @@ int forkpty(int *, char *, const struct termios *, const struct winsize *);
 
 #include "os/pty_proc_unix.c.generated.h"
 
-#if defined(__sun) && !defined(HAVE_FORKPTY)
+#if (defined(__sun) && !defined(HAVE_FORKPTY)) || defined(__HAIKU__)
```

Comment:
ideally we would define a `HAVE_XX` that describes the semantic "feature family" being guarded, so that we don't have zillions of platform-specific decision trees throughout the codebase. if that doesn't make sense for this PR then can make an exception.